src/fluree/db/dbfunctions/internal.cljc

```diff
@@ -579,3 +579,34 @@
     (let [base (.nextDouble (java.util.Random. instant))
           num  (int (Math/floor (* base max')))] num)
     (catch* e (function-error e "rand" instant max'))))
+
+(defn cas
+  "Returns new-val if existing-val is equal to compare-val, else throws exception"
+  [?ctx compare-val new-val]
+  (go-try
+    (let [{:keys [sid pid db]} ?ctx
```

Comment:
you can get rid of this line by destructuring the argument declaration instead:

```clojure
(defn cas
  [{:keys [sid pid db] :as ?ctx} compare-val new-val]
  ...
```

Reply:
My concern is the destructuring is an internal function detail, and something that would make the function's arguments less clear to the caller. It would also create another code format issue: ?ctx would then not be used, so technically shouldn't be included with the :as at all - making the arguments even less clear. So I'm trying to choose the lesser of two evils.